tests/test_agent.py

```diff
@@ -442,6 +442,41 @@ async def test_agent_submit_run_aborts_without_raising_if_server_raises_abort(
             "Server sent an abort signal: message"
         )
 
+    async def test_agent_fails_flow_if_get_infrastructure_fails(
+        self, orion_client, deployment, monkeypatch, mock_infrastructure_run
+    ):
+        flow_run = await orion_client.create_flow_run_from_deployment(
+            deployment.id,
+            state=Scheduled(scheduled_time=pendulum.now("utc")),
+        )
+
+        async def bad_infrastructure(self, name):
+            raise ValueError("Bad!")
+
+        monkeypatch.setattr(
+            "prefect.agent.OrionAgent.get_infrastructure", bad_infrastructure
+        )
```

Comment:
For what it's worth, this can just be `agent.get_infrastructure = AsyncMock(side_effect=ValueError("Bad!"))` once the agent is instantiated. It's nice to avoid monkeypatching when feasible, it's easier to find during refactors.

Reply:
Thanks!. I applied this suggestion.